keras/callbacks.py

```diff
@@ -856,7 +864,50 @@ def on_epoch_end(self, epoch, logs=None):
                     summary_str = result[0]
                     self.writer.add_summary(summary_str, epoch)
                     i += self.batch_size
+        if self.validation_generator and self.histogram_freq:
+            assert self.validation_steps is not None
+            if epoch % self.histogram_freq == 0:
+
+                tensors = (self.model.inputs +
+                           self.model.targets +
+                           self.model.sample_weights)
 
+                i = 0
+                while i < self.validation_steps:
+                    generator_output = next(self.validation_generator)
+                    if len(generator_output) == 2:
+                        x, y = generator_output
+                        sample_weight = None
+                    elif len(generator_output) == 3:
+                        x, y, sample_weight = generator_output
+
+                    if isinstance(x, list):
+                        batch_size = x[0].shape[0]
+                    elif isinstance(x, dict):
+                        batch_size = list(x.values())[0].shape[0]
+                    else:
+                        batch_size = x.shape[0]
+                    if batch_size == 0:
+                        raise ValueError('Received an empty batch. '
+                                         'Batches should at least contain one '
+                                         'item.')
+                    x, y, sample_weight = self.model._standardize_user_data(
+                        x, y, sample_weight=sample_weight,
+                        batch_size=batch_size)
+                    batch_val = x + y + sample_weight
+                    if not len(batch_val) == len(tensors):
+                        raise ValueError('validation_data generator\'s output '
```

Comment:
I am afraid this would not be a very helpful or understandable error message. What specific user error is it meant to catch?

Reply:
I copied this assertion from ~40 lines obove, line 861:
`len(batch_val) == len(tensors)`
that assertion is also done per batch in the part of code that handles preloaded validaton data (i.e. not from a generator).
The user error this assertion (and so do my assertion) meant to catch is when the user given validation generator yields the wrong number of objects, so it cannot be fed into the model.

If you don't agree with this assertion, I would suggest remove the previous one from line 861 as well.